project/contracts/RandomRarity.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@chainlink/contracts/src/v0.8/VRFConsumerBase.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import "hardhat/console.sol";
+
+/**
+    @dev Simple smart contract that allows you to create NFT collections.
+    You create the collection and add items (NFTs) to the collection.
+    Then you can query all metadata of a certain collection via 1 call.
+    Notes:
+    -   1 NFTAlbum (Owned by a user) -> 1..N Collections
+    -   1 Collection -> 1..N Items
+ */
+contract RandomRarity is VRFConsumerBase {
+
+    bytes32 internal keyHash;
+    uint256 internal fee;           // link token fee for oracle
+    uint256 internal randomResult;
+
+    enum Rarity{ 
+        COMMON, 
+        RARE, 
+        EPIC, 
+        LEGENDARY 
+    }
+
+    // numbers are out of 10,000 so we can compare ints
+    uint constant commonChance = 7223;
+    uint constant rareChance = 2200;
+    uint constant epicChance = 457;
+    uint constant legendaryChance = 12;
+    uint256 constant uint256Max = 2**35 - 1;
+
+    /**
+     * Constructor inherits VRFConsumerBase
+     * 
+     * Network: Ropsten Testnet
+     * Chainlink VRF Coordinator address: 0xf720CF1B963e0e7bE9F58fd471EFa67e7bF00cfb
+     * LINK token address:                0x20fE562d797A42Dcb3399062AE9546cd06f63280
+     * Key Hash: 0xced103054e349b8dfb51352f0f8fa9b5d20dde3d06f9f43cb2b85bc64b238205
+     */
+    constructor()
+            VRFConsumerBase(
+            0xf720CF1B963e0e7bE9F58fd471EFa67e7bF00cfb, // VRF Coordinator
```

Comment:
These hashes are specific to the ropsten test net, if we chanve testnet, we need to change these hashes